### PR TITLE
feat: Add injection option for WebContext

### PR DESCRIPTION
**Pull request by acezard**
Provides multiple ways to instantiate the provider, either by props or by self-creating a service like before

### CODE REVIEW

File: packages/cozy-intent/package.json [hunk 1]
```diff
@@ -15,9 +15,12 @@
   },
   "devDependencies": {
     "@babel/core": "7.12.3",
+    "@testing-library/react": "10.4.9",
     "babel-preset-cozy-app": "^2.0.1",
     "eslint-config-cozy-app": "^4.0.0",
-    "react": "17.0.2",
+    "jest": "26.2.2",
+    "mutationobserver-shim": "^0.3.7",
```
Comment (trollepierre) [inline packages/cozy-intent/package.json:22]: i'm lost with all the different PR to review. I already add some reviews comments on another PRs. Could you close one of the two 🙏 ?

**Review by trollepierre — CHANGES_REQUESTED** [review]

File: packages/cozy-intent/package.json (hunk 1 above)
Comment (Ldoppea) [reply packages/cozy-intent/package.json:22]: That's my fault, I rebased my other PR on this branch and I forgot to move the Github pointer to it. So there were too many diffs. Your comments on https://github.com/cozy/cozy-libs/pull/1432 are related to this actual PR.

File: packages/cozy-intent/package.json [hunk 2]
```diff
@@ -40,12 +43,13 @@
...
-    "build": "rm -rf ./dist && yarn build:types && babel --extensions .ts,.tsx ./src -d ./dist --copy-files",
-    "build:types": "tsc -p tsconfig.json",
+    "build": "rm -rf ./dist && yarn build:types && babel --extensions .ts,.tsx --ignore 'src/tests','**/*.spec.tsx','**/*.spec.ts' ./src -d ./dist",
+    "build:types": "tsc -p tsconfig-build.json",
     "build:watch": "yarn build --watch",
     "lint": "eslint . --ext .ts,.tsx",
     "lint:fix": "yarn lint --fix",
-    "prepublishOnly": "yarn build"
+    "prepublishOnly": "yarn build",
+    "test": "jest"
```
Comment (trollepierre) [inline packages/cozy-intent/package.json:52]: after yarn install cozy-libs:
Capture d’écran 2022-01-28 à 11 00 47

File: packages/cozy-intent/src/view/components/WebviewIntentProvider.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+// eslint-disable-next-line no-global-assign
+window = mockWebviewWindow()
+
+jest.mock('cozy-device-helper', () => ({
+  isFlagshipApp: jest.fn()
+}))
+
+jest.mock('post-me', () => ({
+  ChildHandshake: jest.fn(() => Promise.resolve(mockConnection))
```
Comment (trollepierre) [inline packages/cozy-intent/src/view/components/WebviewIntentProvider.spec.tsx:22]:
```suggestion
  ChildHandshake: jest.fn().mockResolvedValue(mockConnection)
```

File: packages/cozy-intent/src/view/components/WebviewIntentProvider.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+}))
+
+describe('WebviewIntentProvider', () => {
+  beforeEach(() => {
+    mockIsFlagshipApp.mockImplementation(() => true)
```
Comment (trollepierre) [inline packages/cozy-intent/src/view/components/WebviewIntentProvider.spec.tsx:27]:
```suggestion
    mockIsFlagshipApp.mockReturnValue(true)
```

File: packages/cozy-intent/src/view/components/WebviewIntentProvider.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  })
+
+  afterEach(() => {
+    mockChildHandshake.mockClear()
+    mockIsFlagshipApp.mockClear()
+  })
+
+  it('renders with no AA context', async () => {
+    mockIsFlagshipApp.mockImplementation(() => false)
```
Comment (trollepierre) [inline packages/cozy-intent/src/view/components/WebviewIntentProvider.spec.tsx:36]:
```suggestion
    mockIsFlagshipApp.mockReturnValue(false)
```

File: packages/cozy-intent/src/view/components/WebviewIntentProvider.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+    render(<WebviewIntentProvider>Hello</WebviewIntentProvider>)
```
Comment (trollepierre) [inline packages/cozy-intent/src/view/components/WebviewIntentProvider.spec.tsx:38]:
```suggestion
    const { queryByText } = render(<WebviewIntentProvider>Hello</WebviewIntentProvider>)
```

File: packages/cozy-intent/src/view/components/WebviewIntentProvider.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+    expect(await screen.findByText('Hello')).toBeDefined()
```
Comment (trollepierre) [inline packages/cozy-intent/src/view/components/WebviewIntentProvider.spec.tsx:40]:
```suggestion
    expect(queryByText('Hello')).toBeDefined()
```

Comment (trollepierre) [reply packages/cozy-intent/src/view/components/WebviewIntentProvider.spec.tsx:40]: apply this suggestion for the 3 tests please

File: packages/cozy-intent/src/view/components/WebviewIntentProvider.spec.tsx [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+
+  it('renders with injection', async () => {
+    render(
+      <WebviewIntentProvider webviewService={mockWebviewService}>
+        Hello
+      </WebviewIntentProvider>
+    )
+
+    expect(await screen.findByText('Hello')).toBeDefined()
+    expect(mockChildHandshake).toBeCalledTimes(0)
```
Comment (trollepierre) [inline packages/cozy-intent/src/view/components/WebviewIntentProvider.spec.tsx:58]:
```suggestion
    expect(mockChildHandshake).not.toHaveBeenCalled()
```

File: packages/cozy-intent/src/view/components/WebviewIntentProvider.spec.tsx [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+    render(<WebviewIntentProvider>Hello</WebviewIntentProvider>)
+
+    expect(await screen.findByText('Hello')).toBeDefined()
+  })
+
+  it('renders without injection', async () => {
+    render(<WebviewIntentProvider>Hello</WebviewIntentProvider>)
+
+    expect(await screen.findByText('Hello')).toBeDefined()
+    expect(mockChildHandshake).toBeCalledTimes(1)
```
Comment (trollepierre) [inline packages/cozy-intent/src/view/components/WebviewIntentProvider.spec.tsx:47]:
```suggestion
    expect(mockChildHandshake).toHaveBeenCalledWith('something')
```

**Review by trollepierre — CHANGES_REQUESTED** [review]

**Review by trollepierre — COMMENTED** [review]
WONDERFUL JOB DONE HERE! I am very excited to see TS implemented with good tests 🎉

File: packages/cozy-intent/package.json (hunk 1 above)
Comment (trollepierre) [reply packages/cozy-intent/package.json:22]: Capture d’écran 2022-01-28 à 11 23 11

**Review by trollepierre — APPROVED** [review]

**Pull request merged**